row_group.go

```diff
@@ -424,9 +424,14 @@ type emptyRowGroup struct {
 
 func newEmptyRowGroup(schema *Schema) *emptyRowGroup {
 	g := &emptyRowGroup{
-		schema:  schema,
-		columns: make([]emptyColumnChunk, numLeafColumnsOf(schema)),
+		schema: schema,
 	}
+	if schema == nil {
+		return g
+	}
```

Comment:
Allowing the schema to be `nil` is going to be error prone and we may have to edit documentation that stated that the schema should never be `nil` in some cases.

Do you think we could have a placeholder schema value representing the empty schema? (kind of like `struct{}` in Go). That way we can continue to require it to never be `nil`, and panic if we ever access a `nil` schema.